File: Assignment 2/C++/1b.cpp
#include<iostream>
using namespace std;
int main(){
double arr[6][6]={{1,2,3,4,5,6},{1,2,3,4,5,6},{1,2,3,4,5,6},{1,2,3,4,5,6},{1,2,3,4,5,6},{1,2,3,4,5,6}};
int i=0;
int j=0;
double k=0;
int l=0;
int m=0;
for(i=0;i<6;i++){
    for(m=0;m<6;m++){
        for(j=0;j<6;j++){
            for(l=0;l<6;l++){
                if((arr[i][m]-arr[j][l])>0){
                    if(i>j){
                        k=k+arr[i][m]-arr[j][l]+i-j;
                        }
                    else{
                        k=k+arr[i][m]-arr[j][l]+j-i;
                        }
                    }
                else{
                    if(i>j){
                        k=k-arr[i][m]+arr[j][l]+i-j;
                        }
                    else{
                        k=k-arr[i][m]+arr[j][l]+j-i;
                        }
                    }
                }
            l=0;
            }
        j=0;
        }
    m=0;
    }
double average=(k)/(36*36);
cout<<"The average distance is:"<<average;
return 0;
}

File: Assignment 2/Python/1b.py
arr=[[1,2,3,4,5,6],[1,2,3,4,5,6],[1,2,3,4,5,6],[1,2,3,4,5,6],[1,2,3,4,5,6],[1,2,3,4,5,6]]
i=0
j=0
k=0
l=0
m=0
while i<6:
    while m<6:
        while j<6:
            while l<6:
                k=k+abs(arr[i][m]-arr[j][l])+abs(i-j)
                l+=1
            l=0    
            j+=1
        j=0    
        m+=1
    m=0
    i+=1
average=(k)/(36*36)
print("The average distance is:",average)
#The output is:
#The avarage distance is:3.88889
